tracing/tracing.go

```diff
@@ -6,51 +6,204 @@ package tracing
 	"github.com/pkg/errors"
-	jaeger "github.com/uber/jaeger-client-go"
-	jaegercfg "github.com/uber/jaeger-client-go/config"
-	jaegerprom "github.com/uber/jaeger-lib/metrics/prometheus"
+	"github.com/uber/jaeger-client-go"
+	jaegerpropagator "go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/contrib/samplers/jaegerremote"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	bridge "go.opentelemetry.io/otel/bridge/opentracing"
+	jaegerotel "go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 )
 
 // ErrInvalidConfiguration is an error to notify client to provide valid trace report agent or config server
 var (
 	ErrBlankTraceConfiguration = errors.New("no trace report agent, config server, or collector endpoint specified")
 )
 
-// installJaeger registers Jaeger as the OpenTracing implementation.
-func installJaeger(serviceName string, cfg *jaegercfg.Configuration, options ...jaegercfg.Option) (io.Closer, error) {
-	metricsFactory := jaegerprom.New()
-
-	// put the metricsFactory earlier so provided options can override it
-	opts := append([]jaegercfg.Option{jaegercfg.Metrics(metricsFactory)}, options...)
-
-	closer, err := cfg.InitGlobalTracer(serviceName, opts...)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize jaeger tracer")
-	}
-	return closer, nil
-}
+const (
+	envSamplerParam           = "JAEGER_SAMPLER_PARAM"
+	envEndpoint               = "JAEGER_ENDPOINT"
+	envJaegerAgentHost        = "JAEGER_AGENT_HOST"
+	envJaegerAgentPort        = "JAEGER_AGENT_PORT"
+	envSamplerType            = "JAEGER_SAMPLER_TYPE"
+	envSamplerRemoteURL       = "JAEGER_SAMPLING_ENDPOINT"
+	DefaultSamplingServerPort = 5778
+)
 
 // NewFromEnv is a convenience function to allow tracing configuration
 // via environment variables
 //
 // Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
 // - JAEGER_AGENT_HOST
-// - JAEGER_SAMPLER_MANAGER_HOST_PORT
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
+// - JAEGER_ENDPOINT
+func NewFromEnv(serviceName string) (io.Closer, error) {
+	cfg, err := parseTracingConfig()
 	if err != nil {
 		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
 	}
-
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
+	if cfg.samplerRemoteURL == "" && cfg.host == "" && cfg.collectorURL == "" {
 		return nil, ErrBlankTraceConfiguration
 	}
+	return cfg.initJaegerTracerProvider(serviceName)
+}
+
+// parseAttributes Parse Jaeger tags from env var JAEGER_TAGS, example of TAGs format: key1=value1,key2=${value2:value3} where value2 is an env var
+// and value3 is the default value, which is optional.
+func parseAttributes(sTags string) ([]attribute.KeyValue, error) {
```

Comment:
[nit] I would name it for what it does.

```suggestion
func parseJaegerTags(sTags string) ([]attribute.KeyValue, error) {
```